### ISSUE TITLE
Multiple desktop support

### ISSUE THREAD

**Issue by kud** [sindresorhus/macos-wallpaper#4]
It could be great to work on every desktop. :)

**Comment by SamVerschueren** [comment]
Do you mean #2 ?

**Comment by kud** [comment]
It depends. I don't speak about multiple monitor but only multiple destkops. Maybe it does the same.

**Comment by Industrial** [comment]
I think it's multiple Desktop's, because how desktops are displayed across monitors is handled by OSX :-)

**Comment by micahcowan** [comment]
If this is the same issue I'm thinking of, then to be clear: I'll maybe have several full-screen windows open, or various spaces for operating, and each will be in a different Desktop. In some cases, it's full-screened windows, in which case the wallpaper makes little practical difference (you can still see it, though, behind the menu bar, or if you do the four-fingered "pinch" gesture to bring up the apps menu), In other cases, I have non-fullscreened windows scattered over the backdrop.

These desktops are created (for my gesture settings, at least), by swiping up with four fingers, and a top bar will show the Desktops, which can be added in number by clicking a "plus" icon at the right side of that bar.

The current implementation of **macos-wallpaper** will only set the wallpaper of whichever Desktop I happen to be operating on at the time, leaving the others alone.

It would be nice to have the option to specify numbered Desktops to declare which you want to set, individually, or to have a "current" option (for current behavior), or to specify that all Desktops should be updated (which I suggest as the new default behavior).

`wallpaper.m` has:
```
			bool success = [sw
				setDesktopImageURL:[NSURL fileURLWithPath:args[1]]
				forScreen:screen
				options:so
				error:&err];
```

where `screen` was initialized like:
```
NSScreen *screen = [NSScreen screens].firstObject;
```

I assume that `screens` is where the information we seek could be had. I suppose a "set all desktops" feature would have to iterate for each of those, calling `setDesktopImageURL()` for each?

**Comment by tom-james-watson** [comment]
You can set backgrounds on all spaces with https://github.com/tom-james-watson/wallpaper-macos

**Comment by sindresorhus** [comment]
Closing as there is no easy way to do this for non-persistent command-line tools.

**Comment by sindresorhus** [comment]
Apple radar: https://openradar.appspot.com/radar?id=4998027479613440

**Comment by mhulse** [comment]
I just wanted to say, thanks for writing this code. Additionally, for anyone else that comes across this and feels slightly disappointed about this limitation, one workaround might be to use the native System Preference and have it point to a directory of images:

![image](https://user-images.githubusercontent.com/218624/55679979-e62e2c80-58c8-11e9-9465-c730b205c8d2.png)

In my case, I wrote a Node.js script to generate an image; I wanted to set the desktop dynamically, but I noticed multiple desktops were not getting set and (from what I could tell) the desktop needed to be focused in order for it to work.

I'm willing to let my users know, via the instructions, to just setup their prefs to point at the directory where the images appear … Thinking about it, this gives the end user way more control over how they manage their desktop background pics.

Anyway, just wanted to post this idea for future readers … I didn't think of using the OS's built-in tools until a friend suggested it as an alternative ... I was so fixated on doing things programatically.

Thanks again for this module and for all the research/hard work. This issue just saved me a lot of time.
